core/src/main/java/io/ddf/analytics/AggregationHandler.java

```diff
@@ -122,24 +122,31 @@ public DDF groupBy(List<String> groupedColumns) {
     return this.getDDF();
   }
 
+  protected String buildGroupBySQL(List<String> aggregateFunctions) {
+    String groupedColSql = Joiner.on(",").join(mGroupedColumns);
+
+    String selectFuncSql = convertAggregateFunctionsToSql(aggregateFunctions.get(0));
+    for (int i = 1; i < aggregateFunctions.size(); i++) {
+      selectFuncSql += "," + convertAggregateFunctionsToSql(aggregateFunctions.get(i));
```

Comment:
Good point. This is the legacy code. I just moved it out of the groupby() function for usage in the other places. Let me take a look based on your comment, :)